XML Parser

Charliebarton opened the pull request:
Wrote a class to parse XML file to internal product representation -
needs to have internal category representation clarified before it can
be finished

Charliebarton committed:
XML Parser

Wrote a class to parse XML file to internal product representation -
needs to have internal category representation clarified before it can
be finished

Commit:
Add feature class and Type enum for features

priyeshpatel commented on src/uk/ac/cam/cl/retailcategorymapper/marshalling/XMLParser.java:
```diff
@@ -0,0 +1,221 @@
+	 * we don't what that field processed
+	 * 
+	 * Most of the code was copied from some web-site so I don't really understand
+	 * what it is doing but it seems to work.
```
Was the code you copied under an appropriate license?

Charliebarton replied:
well it was from a tutorial website saying that the founder "love java and open source" so I presume its ok - no formal licensing though

priyeshpatel commented on src/uk/ac/cam/cl/retailcategorymapper/marshalling/XMLParser.java:
```diff
@@ -0,0 +1,221 @@
+	}
+	
+	
+	public static void main(String argv[]) {
```
We can write test cases to test the parser instead of using a main function.

priyeshpatel commented on src/uk/ac/cam/cl/retailcategorymapper/classifier/Type.java:
```diff
@@ -0,0 +1,12 @@
+ *
+ */
+public enum Type {
+	ORIGINALCATEGORY, NAME, DESCRIPTION, PRICE, GOOGLECATEGORY
```
We shouldn't hard code GOOGLECATEGORY as we will have many different taxonomies.

Perhaps DESTINATIONCATEGORY.

Commit:
Start mapping Products to features for classification

Charliebarton committed:
Partial Category Support

Added some functions to category and allowed for parsing of them

Commit:
Update methods in NaiveBayesClassifier. Change Classifier interface  back to match with manual mapping.

Commit:
Merge branch 'XMLToInternal' of https://github.com/group-papa/category-mapper into XMLToInternal

Commit:
Delete Type enum. Replaced with FeatureType

Charliebarton committed:
Taxonomy TXT parser

created a class to parse the taxonomy from a txt file

Charliebarton committed:
Added Descriptor Parse

for some reason file was not included in last commit - this is necessary
for the last commit to work

Charliebarton committed:
Mapping Parser

Added an XML mapping parser

Charliebarton committed:
Added comments to marshalling code

Commit:
Begin Classify method. Add javadocs to NBC class.

Commit:
Forgot return statement, sorry

Commit:
Revert Mapping to protected. Use MappingBuilder

Charliebarton committed:
Feature equals & hashCode

Added equals and hashCode methods to feature

Charliebarton committed:
Refactored feature extraction

Charliebarton committed:
Added Sanitizer

Sanitizer performs some basic substitutions on the names of products

Commit:
Correct classifier formula

priyeshpatel committed:
Merge master and resolve some conflicts

priyeshpatel committed:
Merge changes to entities

priyeshpatel committed:
Moved feature package

priyeshpatel committed:
Improve unmarshalling

priyeshpatel committed:
Tidy formatting
